fondant/cli.py

```diff
@@ -170,40 +68,198 @@ def pipeline_from_string(import_string: str) -> Pipeline:
     return instance
 
 
-@subcommand(
-    "compile",
-    help="Compile a fondant pipeline",
-    args=[
-        argument(
+class FondantCLI:
+    def __init__(self, input_args: t.List[str] = sys.argv[1:]):
+        parser = argparse.ArgumentParser(
+            description="Fondant CLI",
+            add_help=False,
+        )
+        parser.add_argument(
+            "command",
+            choices=["explore", "compile", "run"],
+        )
+        args, x_args = self._help_or_parse(input_args, parser)
+        getattr(self, args.command)(x_args)
+
+    def _help_or_parse(
+        self,
+        input_args: t.List[str],
+        parser: argparse.ArgumentParser,
+    ) -> t.Tuple[argparse.Namespace, t.List[str]]:
+        """Check if the next command is help and print help if it is.
+        Otherwise parse the known args.
+        """
+        if input_args[0] in ["help", "-h", "--help"]:
```

Comment:
Why don't we rely on the built-in behavior of `argparse` here?

Reply:
The `parse_known_args()` will always consume the help. So if you do `fondant compile local --help` you will get the help page for just fondant.